crates/prompt_gateway/src/prompt_stream_context.rs

```diff
@@ -682,6 +603,7 @@ impl StreamContext {
         }
 
         let tool_calls = model_resp.message.tool_calls.as_ref().unwrap();
```

Comment:
So tool calls happens on both filters?

Reply:
not on llm filter, it should only be in prompt gateway